### DISCUSSION TITLE
[BUG] Remove-PnPFileVersion : The request was cancelled by an event receiver.

### DISCUSSION THREAD

**Discussion by jarvann** [pnp/powershell#1445] [General]
### Reporting an Issue or Missing Feature
Reporting an error that I am seeing while trying to delete File Versions from a User OneDrive. No matter which argument configuration I try, I get "The request was cancelled by an event receiver." message from the cmdlet. I have checked online for issue like this, but can find nothing directly related to deleting versions from OneDrive. I doubt it's actually an issue with the cmdlet, but I wanted to post here just in case.

### Expected behavior 
When I run this command: Remove-PnPFileVersion -Url "[PATH TO DOCUMENT]" -IDENTITY ####### -Force
I expect it to delete a file, but it does not. I have run with both -Force and without.

### Actual behavior 
5691392 11116.0      12/10/2021 6:16:46 PM
5691904 11117.0      12/10/2021 6:54:33 PM
5692416 11118.0      12/10/2021 7:27:30 PM
5692928 11119.0      12/10/2021 7:30:51 PM
5693440 11120.0      12/10/2021 7:33:30 PM

PS C:\Users\MYUSERNAME> Remove-PnPFileVersion -Url "Documents/XXX_FOLDER_PATH/YYYY_FILE.db" -Force -Identity 5693440
Remove-PnPFileVersion : The request was cancelled by an event receiver.
At line:1 char:1
+ Remove-PnPFileVersion -Url "Documents/Documents/My Connect IM Recogni ...
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : WriteError: (:) [Remove-PnPFileVersion], ServerException
    + FullyQualifiedErrorId : EXCEPTION,PnP.PowerShell.Commands.Files.RemoveFileVersion
PS C:\Users\MYUSERNAME> Remove-PnPFileVersion -Url "Documents/XXX_FOLDER_PATH/YYYY_FILE.db" -Identity 5693440

Confirm
Remove a version?
[Y] Yes  [N] No  [S] Suspend  [?] Help (default is "Y"): Y
Remove-PnPFileVersion : The request was cancelled by an event receiver.
At line:1 char:1
+ Remove-PnPFileVersion -Url "Documents/XXX_FOLDER_PATH ...
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : WriteError: (:) [Remove-PnPFileVersion], ServerException
    + FullyQualifiedErrorId : EXCEPTION,PnP.PowerShell.Commands.Files.RemoveFileVersion

PS C:\Users\MYUSERNAME> Remove-PnPFileVersion -Url "Documents/XXX_FOLDER_PATH/YYYY_FILE.db" -Identity "11120.0"

Confirm
Remove a version?
[Y] Yes  [N] No  [S] Suspend  [?] Help (default is "Y"): Y
Remove-PnPFileVersion : The request was cancelled by an event receiver.
At line:1 char:1
+ Remove-PnPFileVersion -Url "Documents/XXX_FOLDER_PATH ...
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : WriteError: (:) [Remove-PnPFileVersion], ServerException
    + FullyQualifiedErrorId : EXCEPTION,PnP.PowerShell.Commands.Files.RemoveFileVersion

### Steps to reproduce behavior
Single File ~10MB with 11121 versions is consuming 87.9 GB of site quota. We have retention enabled on the OneDrive, and the default site quota is 100GB, but have increased user to 200GB. Current usage on the site is 39.70 GB free of 200.00 GB. Attempting to delete the file directly ultimately fails with a timeout of 10 retry attempts. I have waited 36 hours, and am attempting to remove the file versions. 

### What is the version of the Cmdlet module you are running?
ModuleType Version    Name                                ExportedCommands
---------- -------    ----                                ----------------
Manifest   1.8.0      PnP.PowerShell                      

### Which operating system/environment are you running PnP PowerShell on?
- [X] Windows
- [ ] Linux
- [ ] MacOS
- [ ] Azure Cloud Shell
- [ ] Azure Functions
- [ ] Other : please specify

**Comment by jarvann** [comment]
I confirmed that this also happens on other OneDrive User Sites with different files.

**Comment by gautamdsheth** [comment]
Going to convert this to a discussion. 
Looks like you may have some event receiver attached which is causing this, not sure.
In any case, this is not coming from the cmdlet.
